File: projetoES3/src/model/domain/Fornecedor.java
package model.domain;

import java.util.List;

public class Fornecedor extends EntidadeDominio {

	private String nomeForn;
	private Produto produtosForn;
	private Produto produtoPadraoForn;
	private String generoForn;
	private String dataNascForn;
	private String cpfForn;
	private String emailForn;
	private double rankingForn;
	private String senhaForn;
	private String confirmacaoSenhaForn;
	private Cartao cartao;
	private List<Telefone> telefones;
	private List<Endereco> enderecos;


	public Fornecedor(String nomeForn, String cpfForn, String generoForn, String dataNascForn, 
			List<Endereco> enderecos, String emailForn, double rankingForn, String senhaForn, 
			String confirmacaoSenhaForn, Cartao cartao, List<Telefone> telefones) {
		this.nomeForn = nomeForn;
		this.cpfForn = cpfForn;
		this.generoForn = generoForn; 
		this.dataNascForn = dataNascForn;
		this.emailForn = emailForn; 
		this.rankingForn = rankingForn; 
		this.senhaForn = senhaForn;
		this.confirmacaoSenhaForn = confirmacaoSenhaForn;
		this.cartao = cartao;
		this.telefones = telefones;
		this.enderecos = enderecos;
	}

	public boolean validarCpf() {
		System.out.println("CPF OK!");
		return true;
	}
	
	public boolean validarDadosObrigatorios() {
		if(enderecos.validarEndereco()) {
			System.out.println("DADOS OK!");
			return true;
		}
		return false;		
	}
	
		public boolean validarExistencias() {
		List<Fornecedor> fornecedores = consultar();
		if(fornecedores == null || fornecedores.isEmpty()) {
			System.out.println("EXISTÊNCIA OK!");
			return true;
		}
		return false;
	}
	
	public boolean complementarDtCadastro() {
		System.out.println("DATA OK!");
		return true;
	}
	
	public List<Fornecedor> consultar() {
		System.out.println("CONSULTA REALIZADA...");
		return null;
	}
	
	
	public String getNomeForn() {
		return nomeForn;
	}


	public void setNomeForn(String nomeForn) {
		this.nomeForn = nomeForn;
	}

	public Produto getProdutosForn() {
		return produtosForn;
	}


	public void setProdutosForn(Produto produtosForn) {
		this.produtosForn = produtosForn;
	}


	public Produto getProdutoPadraoForn() {
		return produtoPadraoForn;
	}


	public void setProdutoPadraoForn(Produto produtoPadraoForn) {
		this.produtoPadraoForn = produtoPadraoForn;
	}


	public String getGeneroForn() {
		return generoForn;
	}


	public void setGeneroForn(String generoForn) {
		this.generoForn = generoForn;
	}


	public String getDataNascForn() {
		return dataNascForn;
	}


	public void setDataNascForn(String dataNascForn) {
		this.dataNascForn = dataNascForn;
	}


	public String getCpfForn() {
		return cpfForn;
	}


	public void setCpfForn(String cpfForn) {
		this.cpfForn = cpfForn;
	}

	public String getEmailForn() {
		return emailForn;
	}


	public void setEmailForn(String emailForn) {
		this.emailForn = emailForn;
	}


	public double getRankingForn() {
		return rankingForn;
	}


	public void setRankingForn(double rankingForn) {
		this.rankingForn = rankingForn;
	}


	public String getSenhaForn() {
		return senhaForn;
	}


	public void setSenhaForn(String senhaForn) {
		this.senhaForn = senhaForn;
	}


	public String getConfirmacaoSenhaForn() {
		return confirmacaoSenhaForn;
	}


	public void setConfirmacaoSenhaForn(String confirmacaoSenhaForn) {
		this.confirmacaoSenhaForn = confirmacaoSenhaForn;
	}


	public Cartao getCartao() {
		return cartao;
	}


	public void setCartao(Cartao cartao) {
		this.cartao = cartao;
	}

	public List<Telefone> getTelefones() {
		return telefones;
	}

	public void setTelefones(List<Telefone> telefones) {
		this.telefones = telefones;
	}

	public List<Endereco> getEnderecos() {
		return enderecos;
	}

	public void setEnderecos(List<Endereco> enderecos) {
		this.enderecos = enderecos;
	}	
	
	
	
}

File: typeScript/src/model/domain/Fornecedor.ts
import {EntidadeDominio} from "./EntidadeDominio";
import {Endereco} from "./Endereco";
import {Telefone} from "./Telefone";
import {Cartao} from "./Cartao";

export class Fornecedor extends EntidadeDominio {
    constructor(
        private nomeForn: string,
        private produtoPadraoForn: string,
        private generoForn: string,
        private senhaForn: string,
        private dataNascForn: string,
        private cpfForn: string,
        private emailForn: string,
        private rankingForn: string,
        private cartaoForn: Cartao[],
        private telefoneForn: Telefone[],
        private enderecoForn: Endereco[],
        private produtosForn: string[],
    ) {
        super();
    }

    private status_ativo: boolean = true


    /*public Fornecedor(nomeForn: string, cpfForn: string, generoForn: string, dataNascForn: string,
                      endereco: Endereco, telefone: Telefone, emailForn: string, rankingForn: number, senhaForn: string,
                      confirmacaoSenhaForn: string, cartao: Cartao) {

        this.nomeForn = nomeForn;
        this.cpfForn = cpfForn;
        this.generoForn = generoForn;
        this.dataNascForn = dataNascForn;
//		Endereco endereco =
//		Telefone telefone,
        this.emailForn = emailForn;
        this.rankingForn = rankingForn;
        this.senhaForn = senhaForn;
        this.confirmacaoSenhaForn = confirmacaoSenhaForn;
        //Cartao cartao

    }*/

    getProdutos() {
        return this.produtosForn;
    }

    addProduto(produto: string) {
        this.produtosForn.push(produto);
    }

    getCartoes() {
        return this.cartaoForn;
    }

    addCartao(cartao: Cartao) {
        this.cartaoForn.push(cartao);
    }

    getTelefones() {
        return this.telefoneForn;
    }

    addTelefones(telefones: Telefone) {
        this.telefoneForn.push(telefones);
    }

    getEnderecos() {
        return this.enderecoForn;
    }

    addEnderecos(endereco: Endereco) {
        this.enderecoForn.push(endereco);
    }

    clearEnderecos() {
        this.enderecoForn = []
    }

    getName() {
        return this.nomeForn
    }

    getCpf() {
        return this.cpfForn
    }

    getData() {
        return this.dataNascForn
    }

    getSenha() {
        return this.senhaForn
    }

    getGenero() {
        return this.generoForn
    }

    getProdutoPad() {
        return this.produtoPadraoForn
    }

    getEmail() {
        return this.emailForn
    }

    getRank() {
        return this.rankingForn
    }

    getStatus() {
        return this.status_ativo
    }

    setStatus(status: boolean): void {
        this.status_ativo = status;
    }
}